swift/apple/FirezoneKit/Sources/FirezoneKit/Managers/VPNConfigurationManager.swift

```diff
@@ -76,55 +75,59 @@ public class VPNConfigurationManager {
 
   // Firezone 1.4.14 and below stored some app configuration in the VPN provider configuration fields. This has since
   // been moved to a dedicated UserDefaults-backed persistent store.
-  func migrateConfigurationIfNeeded() async throws {
+  func maybeMigrateConfiguration() async throws {
     guard let protocolConfiguration = manager.protocolConfiguration as? NETunnelProviderProtocol,
-          let providerConfiguration = protocolConfiguration.providerConfiguration as? [String: String]
+          let providerConfiguration = protocolConfiguration.providerConfiguration as? [String: String],
+          let session = session()
     else { return }
 
+    let ipcClient = IPCClient(session: session)
+
     var migrated = false
 
     if let apiURLString = providerConfiguration["apiURL"],
        let apiURL = URL(string: apiURLString),
        apiURL.host != nil,
        ["wss", "ws"].contains(apiURL.scheme) {
-      Configuration.shared.apiURL = apiURL
+      try await ipcClient.setApiURL(apiURL)
```

Comment:
Instead of using the ipc client here directly, would it make sense to still use `Configuration` as a facade here so we have an API here that is the same between MacOS and iOS? If the MacOS config instance has the client as a field, you should be able to effectively hide that.

Reply:
Well this class is used on both sides of the Ipc interface primarily for serialization.

There is no iOS or macOS specific code path here - it works the sane way for each.

Reply:
I wanted to keep the `Configuration` class conceptually simple. I could add another class here but I think just using the internal ipc client in vpnManager keeps this encapsulated because (like with DB migrations) we don't want other changes in the codebase to have side effects here.

We'll be able to remove this in a few months presumably.